design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -104,24 +99,15 @@ type DeleteNegotiableQuotesOutput {
     # Implementation Note: We don't make the deleted quotes accessible
     # because "deleted" means they're hidden from the storefront UI. They will
     # still be visible (and labeled as deleted) for a Seller in the admin
-    negotiable_quotes(
```

Comment:
I see that we have inconsistency on how we design output type in similar cases. Some output types [return boolean](https://github.com/magento/architecture/pull/410/files#diff-1f75a7597d5d9a4c23b7b09139bdde27R193) field, others [allows to query](https://github.com/magento/architecture/pull/418/files#diff-929518073d3c60214efac263e7df1d8cR22) items. Looking into how can we align this. Ability to query items seems more convenient, why do you prefer returning a boolean?

Reply:
Here are the pros and cons that I see.

```
type DeleteNegotiableQuotesOutput {
    negotiable_quotes(
        filter: NegotiableQuoteFilterInput,	
        pageSize: Int = 20,	
        currentPage: Int = 1	
    ): NegotiableQuotesOutput @doc(description: "List of negotiable quotes available to customer")	
}
```

We allow to query remaining items, which might be very helpful for updating the list of the items on the client side after mutation is finished. It would be easier to avoid issue of having a stale data. If developer doesn’t want to update the list from the output of this query some data will have to be retrieved anyways and ignored, which is ok.

```
type DeleteNegotiableQuotesOutput {
    successful: Boolean!
}
```

Need to check if we get false in case of an error or it will return an error (not for this specific mutation, but for similar mutations we have in the code). I would expect it to return false in case of an error, but it's not very convenient for a developer. You also can’t update list on the client side with output from the server with out making a separate query.

Other options. Not in our code base, but I have seen other APIs, return list of removed items and sometimes count. I like this a little better than second option, but the first option I like the most.

I would propose to go with the first option and thing about applying this pattern in other places. What do you think @DrewML, @RakeshJesadiya?